cirq-aqt/cirq_aqt/aqt_sampler.py

```diff
@@ -100,27 +225,64 @@ def _generate_json(
         json_str = json.dumps(seq_list)
         return json_str
 
+    def _parse_legacy_circuit_json(self, json_str: str) -> list[Operation]:
+        """Converts a legacy JSON circuit representation.
+
+        Converts a JSON created for the legacy API into one that will work
+        with the Arnica v1 API.
+
+        Args:
+            json_str: A JSON-formatted string that could be used as the
+                data parameter in the body of a request to the old AQT API.
+        """
+        circuit = []
+        number_of_measurements = 0
+        instruction: Operation
+
+        for legacy_op in json.loads(json_str):
+            if number_of_measurements > 0:
+                raise ValueError("Need exactly one `MEASURE` operation at the end of the circuit.")
+
+            if legacy_op[0] == "Z":
+                instruction = GateRZ(operation="RZ", qubit=legacy_op[2][0], phi=legacy_op[1])
+
+            elif legacy_op[0] == "R":
+                instruction = GateR(
+                    operation="R", qubit=legacy_op[3][0], theta=legacy_op[1], phi=legacy_op[2]
+                )
+
+            elif legacy_op[0] == "MS":
+                instruction = GateRXX(operation="RXX", qubits=legacy_op[2], theta=legacy_op[1])
+
+            elif legacy_op[0] == "Meas":
+                instruction = Measure(operation="MEASURE")
```

Comment:
Consider making these operation names constants, so that, if they are changed, it is centralized to one place.

Reply:
Sensible - have added new Enum for those strings.